tests/dragonfly/cluster_mgr_test.py

```diff
@@ -0,0 +1,162 @@
+    client = aioredis.RedisCluster(decode_responses=True, host="127.0.0.1", port=masters[0].port)
+    await insert_cluster_data(client)
+    await check_cluster_data(client)
+
+    # Migrate ~half of the slots to node 1
+    assert run_cluster_mgr(
+        [
+            f"--action=migrate",
+            f"--target_port={BASE_PORT + 1}",
+            f"--slot_start=8000",
+            f"--slot_end=16383",
+        ]
+    )
+    await check_cluster_data(client)
+
+    # Can only detach node 2 (with no assigned slots)
+    assert not run_cluster_mgr(["--action=detach", f"--target_port={BASE_PORT}"])
+    assert not run_cluster_mgr(["--action=detach", f"--target_port={BASE_PORT + 1}"])
+    assert run_cluster_mgr(["--action=detach", f"--target_port={BASE_PORT + 2}"])
+    await check_cluster_data(client)
+
+    # Can't attach non-replica as replica
+    assert not run_cluster_mgr(
+        [
+            f"--action=attach",
+            f"--target_port={BASE_PORT}",
+            f"--attach_port={BASE_PORT+2}",
+            f"--attach_as_replica=True",
+        ]
+    )
+
+    # Reattach node 2 and migrate some slots to it
+    assert run_cluster_mgr(
+        ["--action=attach", f"--target_port={BASE_PORT}", f"--attach_port={BASE_PORT+2}"]
+    )
+    await check_cluster_data(client)
+    # Can't migrate from 2 nodes
+    assert not run_cluster_mgr(
+        [
+            f"--action=migrate",
+            f"--target_port={BASE_PORT + 1}",
+            f"--slot_start=7000",
+            f"--slot_end=9000",
+        ]
+    )
+    assert run_cluster_mgr(
+        ["--action=migrate", f"--target_port={BASE_PORT + 2}", "--slot_start=0", "--slot_end=2000"]
+    )
+    await check_cluster_data(client)
+    assert run_cluster_mgr(
+        [
+            f"--action=migrate",
+            f"--target_port={BASE_PORT + 2}",
+            f"--slot_start=8000",
+            f"--slot_end=10000",
+        ]
+    )
+    await check_cluster_data(client)
+
+    # Can't attach replica before running REPLICAOF
+    assert not run_cluster_mgr(
+        [
+            f"--action=attach",
+            f"--attach_as_replica=True",
+            f"--target_port={BASE_PORT}",
+            f"--attach_port={replicas[0].port}",
+        ]
+    )
+
+    # Add replicas
+    replica_clients = [replica.client() for replica in replicas]
+    for i in range(NODES):
+        await replica_clients[i].execute_command(f"replicaof localhost {masters[i].port}")
+        assert run_cluster_mgr(
+            [
+                f"--action=attach",
+                f"--attach_as_replica=True",
+                f"--target_port={masters[i].port}",
+                f"--attach_port={replicas[i].port}",
+            ]
+        )
+
+    # Can't take over when target is a master
+    assert not run_cluster_mgr(["--action=takeover", f"--target_port={masters[i].port}"])
+
+    # Take over replica 0
+    assert run_cluster_mgr(["--action=takeover", f"--target_port={replicas[0].port}"])
```

Comment:
takeover action removes it from config?

Reply:
takeover removes (detaches) the previous master from the cluster
it also prints that to the console so that the user is aware:
https://github.com/dragonflydb/dragonfly/blob/main/tools/cluster_mgr.py#L272
and
https://github.com/dragonflydb/dragonfly/blob/main/tools/cluster_mgr.py#L479-L484